### PR TITLE
Removed GNU specific grep parameters to work on more *NIX platforms.

### PR DESCRIPTION
**Pull request by noamik**
Hi anapsix,

I was trying to use your template and scripts for my own purposes and stumbled across some difficulties using it on SmartOS. The reason was that my default grep was a non-GNU version and the GNU version I had was compiled without perl regexes. I diged some deeper and found a solution without using those regexes. Thanks a lot for the effort you have put into this so far. You saved me quite some time.

### CODE REVIEW

File: haproxy_discovery.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #  global
 #  stats socket /run/haproxy/info.sock  mode 666 level user
 
-HAPROXY_SOCK=$(echo $1 | grep -Po "^/[^\s]+" || echo "/var/run/haproxy/info.sock")
+HAPROXY_SOCK=$(echo $1 | tr -d '\040\011\012\015' || echo "/var/run/haproxy/info.sock")
```
Comment (anapsix) [inline haproxy_discovery.sh]: unfortunately, `tr` doesn't return non-zero error code when pattern doesn't match :( so it breaks the functionality when no explicit HAPROXY_SOCK is passed as first argument.. 
we'll have to do it a bit differently here if we want to use `tr`
how about this @noamik? does this work for you?

```
HAPROXY_SOCK="/var/run/haproxy/info.sock"
[ -n "$1" ] && echo $1 | grep ^/ >/dev/null && HAPROXY_SOCK="$(echo $1 | tr -d '\040\011\012\015')"
```

Comment (noamik) [inline haproxy_discovery.sh]: Yes, this is working perfectly. Sorry about not regression-testing the without parameter functionality.

**Comment by anapsix** [comment]
If that works for you @noamik, please update and resubmit the PR..
Thanks!

**Comment by anapsix** [comment]
:+1:  thanks!

**Pull request merged**